### PR TITLE
Feature/tauri backup strategy

### LINKED ISSUE #256: Backup Strategie Tauri
📦 Datenstruktur & Speicherorte
Trenne Applikation und Daten strikt:

App-Ordner: C:\Program Files\Billino\ (wird bei Updates überschrieben)
Daten-Ordner: C:\Users\[User]\AppData\Roaming\Billino\ (bleibt bei Updates erhalten)
data\billino.db - SQLite Datenbank
pdfs\invoices\ - Alle Rechnungs-PDFs
pdfs\summary_invoices\ - Sammelrechnungs-PDFs
backups\ - Automatische Backups
🛡️ Backup-Strategie (3-Stufen-Sicherheit)
Stufe 1: Automatische lokale Backups
Täglich: Automatisches Backup beim App-Start oder Shutdown
Bei jeder Rechnung: Sofort-Backup nach Rechnungserstellung
Struktur in backups\:
daily\billino_2025-12-25.db
invoices\invoice_2025-001.pdf (Mirror aller PDFs)
Behalte letzte 30 Tage
Stufe 2: Doppelte PDF-Speicherung
Primär: pdfs\invoices\invoice_2025-001.pdf
Backup: pdfs\backup_archive\invoice_2025-001.pdf
PDFs werden sofort nach Erstellung an beide Orte geschrieben
Zusätzlich: PDF-Binärdaten in der DB (stored_pdf Tabelle) - hast du ja schon!
Stufe 3: Export-Funktionalität
Manueller "Vollständiger Backup erstellen" Button
Erstellt ZIP mit:
Komplette DB
Alle PDFs
Metadaten (Version, Datum, Rechnungsnummern-Range)
Speicherort: User wählt selbst (USB, Cloud, etc.)
🔄 Update-Strategie (Zero Data Loss)
Vor dem Update:
Automatisches Pre-Update-Backup

App erstellt Snapshot: backups\pre_update_v1.2.3\
Enthält DB + alle PDFs
Version wird getaggt
Migrations-Skripte

Jede App-Version bringt DB-Migrations mit
SQLite: Alembic oder eigene Migration-Scripts
Format: migrations\v1.0.0_to_v1.1.0.sql
Während des Updates:
Tauri Updater nutzt separates Verzeichnis

Neue App-Binaries gehen nach C:\Program Files\Billino\
AppData\Roaming\Billino\ wird nicht angerührt
Beim ersten Start nach Update:

App erkennt DB-Version (z.B. Tabelle schema_version)
Führt nötige Migrationen aus
Bei Fehler: Rollback auf Pre-Update-Backup
Nach dem Update:
Verifikation
App prüft DB-Integrität
Zählt Rechnungen (muss identisch sein)
Prüft ob alle PDFs noch existieren
🚨 Worst-Case Recovery
Szenario: DB ist korrupt
Letzte funktionierende DB aus backups\daily\ wiederherstellen
PDFs sind unabhängig und vollständig vorhanden
Fehlende Rechnungen (zwischen Backup und Korruption) können aus PDFs rekonstruiert werden (OCR/Manuell)
Szenario: Festplatte crasht
User hat hoffentlich manuellen Export gemacht
Wenn nicht: Cloud-Sync (optional, siehe unten)
💡 Zusätzliche Empfehlungen
Optional: Cloud-Sync Integration

Nicht automatisch, sondern als Feature
User kann wählen: Dropbox/OneDrive/Google Drive Ordner
backups\ Ordner wird dort gespiegelt
Rechtliche Absicherung

Deutschland: Rechnungen müssen 10 Jahre aufbewahrt werden
PDFs sind das rechtsgültige Format
DB ist nur Index/Verwaltung
Logging

Jedes Backup wird geloggt: logs\backup.log
Bei Fehler: User-Warnung sofort anzeigen
UI-Features

Dashboard: "Letzte Backup: vor 2 Stunden ✓"
"Backup jetzt erstellen" Button
"Export für Steuerberater" (ZIP mit allen Rechnungen + CSV)
🎯 Konkrete Umsetzung für Tauri
Tauri App Data Structure:
└── AppData/Roaming/Billino/
    ├── data/
    │   └── billino.db (mit stored_pdf Tabelle!)
    ├── pdfs/
    │   ├── invoices/ (primär)
    │   ├── summary_invoices/
    │   └── backup_archive/ (mirror)
    ├── backups/
    │   ├── daily/
    │   ├── pre_update/
    │   └── manual/
    └── logs/

Wichtigste Regel: Bei jeder Rechnungserstellung → 3x speichern:

DB Eintrag
PDF in pdfs/invoices/
PDF in pdfs/backup_archive/

### PR DESCRIPTION
**Pull request by wontknow**
## Zusammenfassung
(Was wurde gemacht? Warum?)

## Verwandte Issues
Closes #256

## Änderungen
- ...
- ...

## Wie testen?
1. Backend starten: ...
2. Frontend: ...
3. Erwartet: ...

## Screenshots / Logs
(bei UI)

## Checkliste (DoD)
- [ ] Funktionalität erfüllt (Akzeptanzkriterien)
- [ ] Readme/Docs aktualisiert (falls nötig)
- [ ] Keine Secrets/Keys committed
- [ ] Lint/Build grün
- [ ] Manuelle Tests (happy path & edge cases)
- [ ] CORS geprüft (Dev/Prod)  
- [ ] PDF erzeugt & visuell geprüft  
- [ ] (optional) E-Rechnung Validierung dokumentiert

### CODE REVIEW

File: backend/main.py [hunk 1]
```diff
@@ -166,6 +166,10 @@ async def lifespan(app: FastAPI):
     logger.info("📊 App marked as not ready (rejecting new requests)")
 
     try:
+        logger.info("💾 Running shutdown backup (DB + PDFs)...")
+        shutdown_backup_result = BackupScheduler.backup_on_shutdown()
+        logger.info(f"💾 Shutdown backup result: {shutdown_backup_result}")
+
```
[the remark this answers is not in the document]
Comment (wontknow) [reply backend/main.py]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/wontknow/Billino/pull/298#discussion_r2718727561)

**Comment by wontknow** [comment]
obsolet, ersetzt durch  #312
